traitlets/config/tests/test_application.py

```diff
@@ -66,7 +66,7 @@ class MyApp(Application):
 
     aliases = Dict({
                     ('fooi', 'i') : 'Foo.i',
-                    ('j', 'fooj') : 'Foo.j',
+                    ('j', 'fooj') : ('Foo.j', "`j` terse help msg"),
```

Comment:
The point is `foo --help` and `foo --help-all` to print gradually more text for options.

Optimally, with `--help`, each option "should" occupy a single line, as most unix-utils do, to waste as little screen estate as possible (ok, we're cannot achieve 1-line bc there are default values, equivalent flags, etc). That is important for apps with a lot of aliases.

The `--help-all` is like printing the full man-page.

Of course it is still possible to use the original help-string.
Personally I use a little function that puts in the tuple the 1st line from the full help-msg.

More detailed reasoning on the PR OP.